h3/src/connection.rs

```diff
@@ -69,33 +73,76 @@ pub trait ConnectionState {
     }
 }
 
+#[allow(missing_docs)]
+pub struct AcceptedStreams<C, B>
+where
+    C: quic::Connection<B>,
+    B: Buf,
+{
+    #[allow(missing_docs)]
+    pub wt_uni_streams: Vec<(SessionId, BufRecvStream<C::RecvStream, B>)>,
+}
+
+impl<B, C> Default for AcceptedStreams<C, B>
+where
+    C: quic::Connection<B>,
+    B: Buf,
+{
+    fn default() -> Self {
+        Self {
+            wt_uni_streams: Default::default(),
+        }
+    }
+}
+
+#[allow(missing_docs)]
 pub struct ConnectionInner<C, B>
 where
     C: quic::Connection<B>,
     B: Buf,
 {
     pub(super) shared: SharedStateRef,
-    conn: C,
+    /// TODO: breaking encapsulation just to see if we can get this to work, will fix before merging
+    pub conn: Arc<Mutex<C>>,
```

Comment:
Maybe we could remove this `Arc<Mutex<C>>`.
Some implementations like quinn allow to cheaply clone the connection using a `Mutex` under the hood. 
So we could just add a bound for `C` and require it to be `Clone`.

If a Connection does not support cheaply cloning it could use a `Arc<Mutex<>>` when implementing the quic connection traits.

Reply:
Oh, didn't see that that change was made by us. Thought it was there from the beginning.

Will see if I can revert the change